packages/oscal-react-library/src/components/OSCALCatalogManageGroup.tsx

```diff
@@ -0,0 +1,1052 @@
+          </Grid>
+        </Container>
+      </OSCALWarningDialog>
+    );
+  };
+  const GroupDialog: React.FC<OSCALGroup> = (data) => {
+    function handleNewGroupClick() {
+      setSelectedItemName("NewGroup");
+    }
+    const selectedGroup = data.group;
+    let ID = "";
+    let Name = "";
+    let preID = "";
+    function handleEditIDChange(event: { target: { value: string | undefined } }) {
+      preID = event.target.value ?? "";
+      ID = preID + "_" + generateUUID();
+    }
+    function handleEditgroupTitleChange(event: { target: { value: string } }) {
+      Name = event.target.value;
+    }
+    function handleAddBelow() {
+      setAddNewGroup(true);
+      setAddBelow(true);
+      setNewGroupParent(selectedGroup);
+      handleSaveNewGroup();
+      setShowCardMenu(false);
+    }
+    function handleIncreaseIndent() {
+      const sibling = selectedGroup.rightSibling;
+      if (sibling === undefined) {
+        setShowCardMenu(false);
+        return;
+      }
+      MoveGroup(selectedGroup.groupID, selectedGroup.groupTitle, sibling.groupID);
+      // Reload the main data
+      getData();
+      setShowCardMenu(false);
+    }
+    function handleDecreaseIndent() {
+      const parentID = selectedGroup.parentID;
+      if (parentID === undefined) {
+        setShowCardMenu(false);
+        return;
+      }
+
+      const parent = orderGroups.find((x) => x.groupID === parentID) ?? defaultGroup;
+      const new_parent_id = parent.parentID;
+
+      MoveGroup(selectedGroup.groupID, selectedGroup.groupTitle, new_parent_id);
+      // Reload the main data
+      getData();
+      setShowCardMenu(false);
+    }
+    function handleSaveNewGroup() {
+      if (edit && !addBelow) {
+        EditGroupTitle(selectedGroup.groupID, Name);
+        setAddNewGroup(false);
+        getData();
+        setEditGroup(defaultGroup);
+        return;
+      }
+      if (Name.length < 1) return;
+      if (preID === "") {
+        ID = Name.substring(0, 1) + ID;
+      }
+      if (addBelow) {
+        const parent_id = newGroupParent === undefined ? "" : newGroupParent.groupID;
+        SaveNewGroup(ID, Name, parent_id);
+        setAddNewGroup(false);
+        getData();
+        setAddBelow(false);
+        return;
+      }
+      SaveNewGroup(ID, Name, "");
+      setAddNewGroup(false);
+      getData();
+    }
+    function handleDeleteGroup() {
+      //setOpenDeleteDialog(true);
+      DeleteGroup(selectedGroup.groupID);
+      getData();
+      setAddNewGroup(false);
+    }
+
+    const isSelected = selectedItemName === "NewGroup" ? true : false;
+    return (
+      <Container
+        onClick={handleNewGroupClick}
+        sx={{ height: 73, border: "1px solid #D2D2D2", background: "#EAEAEA", width: 312 }}
+      >
+        <Container
+          sx={{
+            height: 73,
+            left: 20,
+            position: "absolute",
+            background: "#F6F6F6",
+            border: "1px solid #D2D2D2",
```

Comment:
I would suggest adding `#D2D2D2` and any new occuring colors to our palette theming (in `AppTheme`) if this they are a commonly occurring color (using this color more than 2 times).

Reply:
@Bronstrom, when I define a new theme color in AppTheme.js I ran into an error when I compile the application.
<img width="890" alt="image" src="https://github.com/EasyDynamics/oscal-react-library/assets/59936247/224b0508-db45-4186-bd01-0a5cc7003e5d">
Typescipt do not recognize the newly added theme colors. I do not know if these new colors have to define in another file besides AppTheme.js

Reply:
Oh, I completely forgot. Everytime we define a new `theme` in `AppTheme.js` TypeScript requires us to also define this as a type option for the palette. I created a file called `createPalatte.d.ts` which currently has `backgroundGray` and `primaryAccent` defined currently. We will need to add `lightGray` to the two interfaces there.

Reply:
Was this issue able to be resolved without needing to update `createPalatte.d.ts`?

Reply:
@Bronstrom, Thanks for this suggestion. I added few more palettes to cover all colors we have in the application so far.